pkg/crc/preflight/preflight_checks_darwin.go

```diff
@@ -146,3 +147,27 @@ func fixOcBinaryCached() (bool, error) {
 	logging.Debug("oc binary cached")
 	return true, nil
 }
+
+func checkResolvConfFilePermissions() (bool, error) {
+	err := unix.Access(resolvFile, unix.R_OK|unix.W_OK)
+	if err != nil {
+		return false, fmt.Errorf("%s is not readable/writable by the current user", resolvFile)
+	}
+
+	return true, nil
+}
+
+func fixResolvConfFilePermissions() (bool, error) {
+	logging.DebugF("Making %s readable/writable by the current user", resolvFile)
+	currentUser, err := user.Current()
+	if err != nil {
+		return false, err
+	}
+
+	stdOut, stdErr, err := crcos.RunWithPrivilege("chown", currentUser.Username, resolvFile)
+	if err != nil {
+		return false, fmt.Errorf("Unable to change ownership of the resolver file: %s %v: %s", stdOut, err, stdErr)
+	}
+
+	return true, nil
+}
```

Comment:
You could introduce 2 helpers to share the code with `checkResolverFilePermissions` and `fixResolverFilePermissions`

Reply:
Yes, this refactor I can do 👍